### PR TITLE
MWPW-140071 - CaaS Marquee Metadata Block

### PR DESCRIPTION
**Pull request by sanrai**
- Adds CaaS Marquee Metadata block that builds the correct HTML for the Send To CaaS workflow.

Note: Was going to render a table preview on `.page` but because every other metadata block doesn't I followed that convention (to avoid confusing authors).

Resolves: [MWPW-140071](https://jira.corp.adobe.com/browse/MWPW-140071)

## Test URLs

**Milo Test URLs:**
- [Before](https://main--milo--adobecom.hlx.page/drafts/sanrai/v1-caas-marquee-in-library?martech=off)
- [After](https://caas-marquee-metadata--milo--adobecom.hlx.page/drafts/sanrai/v1-caas-marquee-in-library?martech=off)

### Before
1.  Should say 'Failed loading caas-marquee-metadata block`.
2. Should have Table elements wrongly visible inside pink container.
image
3. Should only render 11 nested divs (not all marquee parameters are being parsed)
image

### After
1. Block should be hidden
image
2. There should be 19 inner nested divs added

The 8 extra fields come from these additional fields being injected.

```
  const additionalFields = {
    arbitrary: ``
    tags: 'caas:content-type/promotion',
    cta1url: `${metadata.cta1url}`,
    cta1text: `${metadata.cta1text}`,
    cta1style: `${metadata.cta1style}`,
    cta2url: `${metadata.cta2url}`,
    cta2text: `${metadata.cta2text}`,
    cta2style: `${metadata.cta2style}`,
  };
```

image

### CODE REVIEW

**Review by Blainegunn — APPROVED** [review]

File: libs/blocks/caas-marquee-metadata/caas-marquee-metadata.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    let val = row.children[1].innerHTML || '';
+    if (key.includes('image')) {
+      const img = row.children[1].querySelector('img');
+      val = img ? new URL(img.src).pathname : '';
+    }
+    if (key.includes('cta')) {
+      metadata = { ...metadata, ...parseCtas(row.children[1]) };
+    }
+    if (key.includes('variant')) {
+      val = val.replace(/,/g, '');
```
Comment (sanrai) [inline libs/blocks/caas-marquee-metadata/caas-marquee-metadata.js:38]: Update to: `val = val.replaceAll(',' , ' ')`

File: libs/blocks/caas-marquee-metadata/caas-marquee-metadata.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    index += 1;
+  }
+  return ctas;
+}
+export const getMetadata = (el) => {
+  let metadata = {};
+  for (const row of el.children) {
+    const key = row.children[0].textContent.trim().toLowerCase() || '';
+    let val = row.children[1].innerHTML || '';
+    if (key.includes('image')) {
```
Comment (sanrai) [inline libs/blocks/caas-marquee-metadata/caas-marquee-metadata.js:30]: update to `key.startsWith('image')`

**Review by cmiqueo — APPROVED** [review]

**Review by arshadparwaiz — APPROVED** [review]

**Comment by sanrai** [comment]
Scheduled to be merged/released on 02/07 at 2 PM PST.

File: tools/send-to-caas/send-utils.js [hunk 3]
```diff
@@ -315,7 +321,7 @@ const getCountryAndLang = async (options) => {
 
 const parseCardMetadata = () => {
   const pageMd = {};
-  const mdEl = getConfig().doc.querySelector('.card-metadata');
+  const mdEl = getConfig().doc.querySelector('.card-metadata') || getConfig().doc.querySelector('.caas-marquee-metadata');
```
Comment (cmiqueo) [inline tools/send-to-caas/send-utils.js:324]: styling:
const mdEl = getConfig().doc.querySelector('.card-metadata') 
    || getConfig().doc.querySelector('.caas-marquee-metadata');

**Pull request merged**